key.go

```diff
@@ -59,18 +60,13 @@ type PrivKey interface {
 
 	// Generate a secret string of bytes
 	GenSecret() []byte
-
-	Decrypt(b []byte) ([]byte, error)
 }
 
 type PubKey interface {
 	Key
 
 	// Verify that 'sig' is the signed hash of 'data'
 	Verify(data []byte, sig []byte) (bool, error)
-
-	// Encrypt data in a way that can be decrypted by a paired private key
-	Encrypt(data []byte) ([]byte, error)
```

Comment:
Yup, NaCL implements it also, it is supposed to be also proven to be secure http://crypto.stackexchange.com/questions/3260/using-same-keypair-for-diffie-hellman-and-signing/3311#3311 which isn't true for RSA.